src/run.c

```diff
@@ -213,55 +253,47 @@ FILE* readback_file(Port port) {
   return fp;
 }
 
-// Reads a single char from `argm`.
-Port io_read_char(Net* net, Book* book, Port argm) {
-  FILE* fp = readback_file(peek(net, argm));
-  if (fp == NULL) {
+// Reads from a file a specified number of bytes.
+// `argm` is a tuple of (file_descriptor, num_bytes).
+Port io_read(Net* net, Book* book, Port argm) {
+  if (get_tag(peek(net, argm)) != CON) {
```

Comment:
`peek` is not enough here and in all the other places. If we pass a constant hidden behind a `REF` node these IO functions will fail.

For example, if in Bend I call open like this:
```
(IO/call IO/MAGIC "OPEN" (MyConstFile, "r") @x x)
```
Where `MyConstFile` is a reference to a function that contains the file path.

Instead of `peek` we should use a different function that has a loop of peek and expand. This has the problem of looping infinitely if the user accidentally passes a recursive ref.

Reply:
switched to using `expand` for every arg of every io func